website/docs/common-scripts/xudt.mdx

```diff
@@ -0,0 +1,346 @@
+
+`is_owner_mode` indicates if the current xUDT is unlocked via owner mode(as
+described by sUDT), `extension_index` refers to the index of the current
+extension in the `ScriptVec` structure. `args` and `args_length` are set to the
+script args included in `Script` structure of the current extension script.
+
+If this function returns 0, the current extension script validation is
+considered successful.
+
+• If `flags & 0x1FFFFFFF` is 0x2, extension data will contain the blake160 hash
+of the `ScriptVec` structure as explained in the previous section. The
+actual `extension_scripts` (`ScriptVec`) structure data will be included in a
+witness field `input_type` or `output_type` contained in the current
+transaction. We will explain this part below. Choosing `input_type` or
+`output_type` depends on whether the type script is running on input or output
+cells. Under a lot of scenarios, it is `input_type`. But in the following example “Owner
+Mode Without Consuming Cell”, we can see it’s possible on `output_type`.
+
+### Witness Structure
+
+The `input_type` or `output_type` field in `WitnessArgs` has the following data
+structure in molecule format:
+
+```js
+table XudtWitness {
+    owner_script: ScriptOpt,
+    owner_signature: BytesOpt,
+    extension_scripts: ScriptVecOpt,
+    extension_data: BytesVec,
+}
+```
+
+The field `owner_script` and `owner_signature` will be used in owner mode. The
+field `extension_scripts` is used when `flags & 0x1FFFFFFF` is 0x2 in args.
+
+The length of `extension_data` structure inside must also be the same as
+`ScriptVec` in `xUDT args` or `extension_scripts`. An extension script might also
+require transaction-specific data for validation. The witness here provides a
+place for these data needs.
+
+### Owner Mode Update
+
+As described in RFC sUDT, if an input cell in the current transaction uses an
+input lock script with the same script hash as the owner lock script hash, the
+`is_owner_mode` will be set to true. In xUDT, this rule is updated by the
+following rule:
+
+If an input or output cell in the current transaction uses one or more of the
+following:
+
+- input lock script (when `flags & 0x20000000` is **zero** or `flags` is not present)
+- output type script (when `flags & 0x40000000` is **non-zero**)
+- input type script (when `flags & 0x80000000` is **non-zero**)
+
+With the same script hash as the owner lock script hash, the `is_owner_mode`
+will be set to true. The output lock scripts are not included, because they
+won’t be run in a transaction.
+
+If the `owner_script` in witness isn’t none and its blake2b hash is the same as
+the owner lock script hash in `args`, this script will be run as an extension
+script. If the script returns success, `is_owner_mode` is set to true. Note, the
+`owner_signature` field can be used by this owner script. When tokens are
+minted, the `owner_script` and `owner_signature` can be set to some proper
+values. When tokens are transferred, they can be set to none.
```

Comment:
```suggestion
With the same Script hash as the owner Lock Script hash, the `is_owner_mode`
will be set to true. The output Lock Scripts are not included because they
won’t be run in a transaction.

If the `owner_script` in witness isn’t none and its blake2b hash matches
the owner Lock Script hash in `args`, this Script will be run as an extension
Script. If the Script returns success, `is_owner_mode` is set to true. Note that the
`owner_signature` field can be used by this owner Script. When tokens are
minted, the `owner_script` and `owner_signature` can be set to appropriate
values. When tokens are transferred, they can be set to none.
```